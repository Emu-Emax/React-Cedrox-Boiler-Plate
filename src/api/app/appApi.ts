import API from '../client'

export const getMockData = () =>
  // const { data }: { data: unknown[] } = await API.get('mock-url')
  []
