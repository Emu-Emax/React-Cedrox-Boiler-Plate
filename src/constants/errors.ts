export const ERROR_API_DEFAULT = 'Something went wrong.'
