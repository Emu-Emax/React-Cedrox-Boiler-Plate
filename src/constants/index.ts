export * from 'constants/app'
export * from 'constants/routes'
