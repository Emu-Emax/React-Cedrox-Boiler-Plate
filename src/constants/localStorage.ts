export const LOCAL_STORAGE_THEME = 'theme'
