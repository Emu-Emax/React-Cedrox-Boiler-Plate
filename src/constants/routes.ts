export const ROUTE_ERROR = '/error'
export const ROUTE_HOME = '/'
