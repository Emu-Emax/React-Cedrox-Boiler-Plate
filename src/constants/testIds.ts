export const TEST_ID_TOOLBAR_LIGHT_DARK_SWITCH_CONTAINER =
  'toolbar-light-dark-switch-container'
export const TEST_ID_TOOLBAR_LIGHT_DARK_SWITCH_BUTTON =
  'toolbar-light-dark-switch-button'
