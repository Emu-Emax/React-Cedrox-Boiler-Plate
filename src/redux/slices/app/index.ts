import { appSlice } from 'redux/slices/app/appSlice'

export * from 'redux/slices/app/appActions'
export * from 'redux/slices/app/appSelectors'

export const { setTheme } = appSlice.actions

export default appSlice.reducer
