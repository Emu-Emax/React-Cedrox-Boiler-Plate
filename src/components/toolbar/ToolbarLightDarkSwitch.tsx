import { useAppDispatch, useAppSelector } from 'hooks/useRedux'
import styled, { css } from 'styled-components'
import { ReactComponent as SunIcon } from 'assets/icons/sunflower-svgrepo-com.svg'
import { ReactComponent as MoonIcon } from 'assets/icons/moon-svgrepo-com.svg'
import { selectAppTheme, setTheme } from 'redux/slices/app'
import {
  TEST_ID_TOOLBAR_LIGHT_DARK_SWITCH_BUTTON,
  TEST_ID_TOOLBAR_LIGHT_DARK_SWITCH_CONTAINER,
} from 'constants/testIds'
import { DARK_THEME, LIGHT_THEME } from 'constants/index'

const ToolbarLightDarkSwitchContainer = styled.div(
  ({ theme }) => css`
    display: flex;
    padding: 2px;
  `
)

const SwitchButton = styled.button<{ currentTheme: string }>(
  ({ theme, currentTheme }) => css`
    background: ${theme.colors.gradient};
    border: 2px solid ${theme.colors.buttonBorder};
    border-radius: 30px;
    cursor: pointer;
    display: flex;
    font-size: 0.5rem;
    justify-content: space-between;
    margin: 0 auto;
    overflow: hidden;
    position: relative;
    width: 5rem;
    height: 3rem;

    svg {
      height: auto;
      width: 2.5rem;
      transition: all 0.3s linear;

      // sun icon
      &:first-child {
        transform: ${currentTheme === 'light'
          ? 'translateY(4px)'
          : 'translateY(100px)'};
      }

      // moon icon
      &:nth-child(2) {
        transform: ${currentTheme === 'light'
          ? 'translateY(-100px)'
          : 'translateY(4px)'};
      }
    }
  `
)

const ToolbarLightDarkSwitch = () => {
  const currentTheme = useAppSelector(selectAppTheme)
  const dispatch = useAppDispatch()

  const handleSwitch = () => {
    const newTheme = currentTheme === LIGHT_THEME ? DARK_THEME : LIGHT_THEME
    dispatch(setTheme(newTheme))
  }

  return (
    <ToolbarLightDarkSwitchContainer
      data-testid={TEST_ID_TOOLBAR_LIGHT_DARK_SWITCH_CONTAINER}
    >
      <SwitchButton
        onClick={handleSwitch}
        currentTheme={currentTheme}
        data-testid={TEST_ID_TOOLBAR_LIGHT_DARK_SWITCH_BUTTON}
      >
        <SunIcon />
        <MoonIcon />
      </SwitchButton>
    </ToolbarLightDarkSwitchContainer>
  )
}

export default ToolbarLightDarkSwitch
