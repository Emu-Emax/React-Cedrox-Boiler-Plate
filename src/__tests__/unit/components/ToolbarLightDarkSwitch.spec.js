import ToolbarLightDarkSwitch from 'components/toolbar/ToolbarLightDarkSwitch';
import { TEST_ID_TOOLBAR_LIGHT_DARK_SWITCH_CONTAINER } from 'constants/testIds';
import { fireEvent, screen } from '@testing-library/react';
import { TEST_ID_TOOLBAR_LIGHT_DARK_SWITCH_BUTTON } from 'constants/testIds';
import * as appPlainActions from 'redux/slices/app';
import { spySetTheme } from '__spies__';

describe('ToolbarLightDarkSwitch', () => {
  it('renders correctly and matches snapshot', () => {
    const { container } = customRender(<ToolbarLightDarkSwitch />);
    expectMatchSnapshot(container);
    expect(
      screen.getByTestId(TEST_ID_TOOLBAR_LIGHT_DARK_SWITCH_CONTAINER)
    ).toBeInTheDocument();
  });

  it('correctly switches themes', () => {
    spySetTheme();
    customRender(<ToolbarLightDarkSwitch />);
    const switchButton = screen.getByTestId(
      TEST_ID_TOOLBAR_LIGHT_DARK_SWITCH_BUTTON
    );
    fireEvent.click(switchButton);
    expect(appPlainActions.setTheme).toHaveBeenCalled();
  });
});
