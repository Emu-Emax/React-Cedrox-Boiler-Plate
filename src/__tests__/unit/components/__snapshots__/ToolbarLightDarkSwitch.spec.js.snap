// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolbarLightDarkSwitch renders correctly 1`] = `
<div>
  <div
    class="sc-bcXHqe cEJzEP"
  >
    <button
      class="sc-gswNZR eGPipM"
    >
      <svg>
        sunflower-svgrepo-com.svg
      </svg>
      <svg>
        moon-svgrepo-com.svg
      </svg>
    </button>
  </div>
</div>
`;

exports[`ToolbarLightDarkSwitch renders correctly and matches snapshot 1`] = `
<div>
  <div
    class="sc-bcXHqe cEJzEP"
    data-testid="toolbar-light-dark-switch-container"
  >
    <button
      class="sc-gswNZR eGPipM"
      data-testid="toolbar-light-dark-switch-button"
    >
      <svg>
        sunflower-svgrepo-com.svg
      </svg>
      <svg>
        moon-svgrepo-com.svg
      </svg>
    </button>
  </div>
</div>
`;
